src/client/api/restful/get_outputs_id.c

```diff
@@ -96,23 +101,32 @@ int deser_outputs(char const *const j_str, res_outputs_id_t *res) {
     goto end;
   }
 
-  if ((ret = json_get_uint32(json_obj, JSON_KEY_LIMIT, &res->u.output_ids->limit) != 0)) {
-    printf("[%s:%d]: gets %s failed\n", __func__, __LINE__, JSON_KEY_LIMIT);
+  if ((ret = json_get_uint64(json_obj, JSON_KEY_LEDGER_IDX, &res->u.output_ids->ledger_idx) != JSON_OK)) {
+    printf("[%s:%d]: gets %s failed\n", __func__, __LINE__, JSON_KEY_LEDGER_IDX);
     goto end;
   }
 
-  if ((ret = json_get_uint32(json_obj, JSON_KEY_COUNT, &res->u.output_ids->count) != 0)) {
-    printf("[%s:%d]: gets %s failed\n", __func__, __LINE__, JSON_KEY_COUNT);
+  if ((ret = json_get_uint32(json_obj, JSON_KEY_PAGE_SIZE, &res->u.output_ids->page_size) != JSON_OK)) {
+    printf("[%s:%d]: gets %s failed\n", __func__, __LINE__, JSON_KEY_PAGE_SIZE);
     goto end;
   }
 
-  if ((ret = json_string_array_to_utarray(json_obj, JSON_KEY_DATA, res->u.output_ids->outputs)) != 0) {
-    printf("[%s:%d]: gets %s failed\n", __func__, __LINE__, JSON_KEY_DATA);
-    goto end;
+  // parse for cursor, it is an optional paramater
+  cJSON *json_cursor = cJSON_GetObjectItemCaseSensitive(json_obj, JSON_KEY_CURSOR);
+  if (json_cursor != NULL) {
+    if (cJSON_IsString(json_cursor) && (json_cursor->valuestring != NULL)) {
+      res->u.output_ids->cursor = malloc(strlen(json_cursor->valuestring) + 1);
+      memset(res->u.output_ids->cursor, 0, strlen(json_cursor->valuestring) + 1);
```

Comment:
```suggestion
      res->u.output_ids->cursor = calloc(1, strlen(json_cursor->valuestring) + 1);
```
Instead of calling two functions, a `calloc` function may be used which allocates memory and initializes it with 0.

Reply:
Used memset as we have not used calloc anywhere in our lib. Changed to calloc, better.